### PR TITLE
Allow setting domain when deleting cookies

### PR DESCRIPTION
**Pull request by alexpdraper**
## Changes

Adds `domain` as a property on the delete cookie options. Without this it wasn’t possible to use `Astro.cookies.delete` to delete a cookie that has the domain set.

## Testing

Added a unit test that checks the domain is added to the set-cookie header when provided, similar to the test for the path option.

## Docs

/cc @withastro/maintainers-docs for feedback!
The API reference for `Astro.cookies.delete` doesn’t include the options as one of the arguments.

https://github.com/withastro/docs/blob/main/src/pages/en/reference/api-reference.md#astrocookies

### CODE REVIEW

File: .changeset/small-ravens-cover.md
Comment (matthewp) [inline .changeset/small-ravens-cover.md:2]: This should be a `minor` since it's a new feature.

**Comment by matthewp** [comment]
The addition makes sense to me. We will indeed need docs before merging this. The docs for that are here: https://github.com/withastro/docs/blob/da3725bf4e3a24647ee240a3816cc5d72e31d9e6/src/pages/en/reference/api-reference.md#astrocookies

**Comment by matthewp** [comment]
Pretty sure the tests failing is unrelated to your change and it's a problem in `main`. I'm looking to fix that, so please disregard.

**Review by matthewp — CHANGES_REQUESTED** [review]
- Should be a `minor` change.
- Needs docs

**Comment by matthewp** [comment]
Thanks @alexpdraper, when you have a docs PR please link to that from here. Then the docs team will review it and approve this PR.

**Comment by sarah11918** [comment]
This looks like a useful feature @alexpdraper! You're on our radar now at docs, and we'll be on the lookout for a PR.

Please PR a suggested update to this section of the docs: https://docs.astro.build/en/reference/api-reference/#astrocookies (file is at: https://github.com/withastro/docs/blob/main/src/pages/en/reference/api-reference.md )

and we will help over there with final wording, formatting, and just generally squishing it in to that table!

**Comment by matthewp** [comment]
Oops, closed by mistake.

**Comment by alexpdraper** [comment]
> @alexpdraper Are you planning on making the docs changes?

Yes, certainly. Sorry about the delay.

@sarah11918 here’s the docs PR https://github.com/withastro/docs/pull/2151

**Comment by sarah11918** [comment]
Docs PR received! Removing Astro-Docs maintainers requirement for review here, and Docs will take this up over there. 🚀

**Review by matthewp — CHANGES_REQUESTED** [review]
Looks good, this will go out with the next minor (later this week)

**Comment by alexpdraper** [comment]
Great! Thanks so much!

**Pull request merged**
